### PR TITLE
Add Project Injury Reaction (PiR) compatibility for plate damage and …

### PR DESCRIPTION
**Pull request by x60secAssassin**
…reactions

Integrates APS plate absorption with PiR HandleDamage, HitPart, and PiRredirect paths.

Adds CBA settings for blocking residual PiR damage while plates hold, optional PiRanim reactions, and optional AI rules.

New functions: installPiRHandleDamage, installPiRHitPart, dispatchPiRReaction, piREffectivePlateSettings, and helpers.

Tested on APS 0.11.2.x with PiR loaded (player + AI scenarios).

## Summary

Adds **optional compatibility** between **Armor Plates System (APS)** and **Project Injury Reaction (PiR)** so plate absorption and PiR injury/reaction logic work together instead of fighting each other.

**Problem:** With PiR loaded, APS plate math can absorb ballistic hits, but PiR may still apply injury through its own `HandleDamage` / `HitPart` / `PiRredirect` paths—often causing instant unconsciousness or full `PiR0` reactions while plates are intact.

**Approach:** When PiR is detected, APS switches to an “external medical” mode and installs coordinated hooks that:
- Run plate armor handling on PiR’s damage pipeline (`installPiRHandleDamage`)
- Intercept PiR `HitPart` / redirect flow where needed (`installPiRHitPart`, wrappers on `PiRredirect` / `PiRredirect0`)
- Centralize reaction dispatch (`dispatchPiRReaction`) so player-like units can use **animation-only** PiR (`PiRanim` / `PiRanim0`) when plates fully absorb a hit, while AI keeps stock PiR behavior unless explicitly enabled

## New CBA settings (player + AI)

| Setting | Purpose |
|--------|---------|
| **Player: Block PiR Damage Until Plates Break** | While plates remain, force residual PiR injury from protected hits toward zero and sync with HitPart handling |
| **Player: Allow PiR Reactions While Plates Hold** | If blocking is on, still play PiR hit reactions (`PiRanim`) on absorbed hits |
| **AI: Use Custom PiR Plate Rules** | Opt-in: apply the custom HitPart/damage bridge to AI |
| **AI: Block PiR Damage Until Plates Break** | AI equivalent of player block (requires AI custom rules) |
| **AI: Allow PiR Reactions While Plates Hold** | AI equivalent of player reactions |

## Main technical changes

**New functions (`addons/main/functions/`):**
- `fnc_installPiRHandleDamage.sqf` — PiR-aware `HandleDamage` EH: plate armor, `lastDamageBridge` state, structural/incapacitated follow-up handling
- `fnc_installPiRHitPart.sqf` — Local `HitPart` EH to route hits through plate rules before PiR redirect
- `fnc_dispatchPiRReaction.sqf` — Chooses `PiR`/`PiR0` vs `PiRanim`/`PiRanim0` based on plate bridge + CBA settings
- `fnc_piREffectivePlateSettings.sqf`, `fnc_shouldSkipPiRRedirect.sqf`, `fnc_shouldSuppressPiRReaction.sqf`, `fnc_pirShansFromHitSelection.sqf`

**Also updates:** `XEH_postInit.sqf`, `XEH_preInit.sqf`, `XEH_PREP.hpp`, `initSettings.inc.sqf`, `initSettingsACE.inc.sqf`, `initSettingsTorsoPlates.inc.sqf`, `stringtable.xml`, and related hooks in existing damage/plate functions.

## Dependencies

- **CBA_A3** (existing APS requirement)
- **Project Injury Reaction (PiR)** — soft dependency; integration only activates when PiR is loaded

Tested on **APS 0.11.2.x** with PiR (including PiR instant-death enabled) in single-player and Zeus scenarios.

## Notes for review

1. **Scope:** Substantial integration in `XEH_postInit` plus dedicated PiR install functions—not a one-line patch. Happy to split into smaller PRs if preferred.
2. **Defaults:** PiR block/reaction/AI rule checkboxes default to **on** in the contributor’s playtest fork; happy to change upstream defaults to your preference.
3. **Edge case:** Rare instant vanilla-style death after heavy plate soak under movement is under investigation (not consistently reproducible).

## Test plan

- [ ] Load `@CBA_A3`, APS, and PiR; confirm PreInit logs PiR external medical mode
- [ ] Player with 3 plates: sustained torso fire — reactions play while plates hold (block + allow reactions on)
- [ ] Last plate breaks — PiR injury/unconsciousness behaves as expected
- [ ] AI with custom PiR plate rules **off** — stock PiR behavior
- [ ] AI with custom rules **on** — plate bridge applies per AI CBA settings

### CODE REVIEW

File: addons/main/ui/skull.psd
Comment (diwako) [inline addons/main/ui/skull.psd:1]: This is a resource file, this must not be deleted.

File: addons/main/LICENSE
Comment (diwako) [inline addons/main/LICENSE:1]: The license is of outmost importance., It must NOT be deleted or changed!

File: addons/main/$PBOPREFIX$
Comment (diwako) [inline addons/main/$PBOPREFIX$:1]: Restore this file please.

File: addons/main/$pboprefix$.txt
Comment (diwako) [inline addons/main/$pboprefix$.txt:1]: Do not use mikero tools in this project. Use hemtt
https://github.com/BrettMayson/HEMTT/

File: addons/main/rap.txt
Comment (diwako) [inline addons/main/rap.txt:1]: not needed file, remove

File: addons/main/texheaders.bin
Comment (diwako) [inline addons/main/texheaders.bin:1]: not needed file, remove

File: addons/main/data/armor_plate.p3d
Comment (diwako) [inline addons/main/data/armor_plate.p3d:1]: Why was this p3d changed? It should stay unbinarized and changeable still.

File: addons/main/CfgWeapons.hpp [hunk 1]
```diff
@@ -6,7 +6,7 @@ class CfgWeapons {
     class GVAR(plate): CBA_MiscItem {
         scope = 2;
         author = "diwako";
-        model = QPATHTOF(data\armor_plate.p3d);
```
Comment (diwako) [inline addons/main/CfgWeapons.hpp:9]: Should stay the custom p3d

File: addons/main/config.cpp
Comment (diwako) [inline addons/main/config.cpp:1]: Do not replace it with the rapified and then unraprified file. Make changes if you need to inside the included files

File: addons/main/stringtable.xml
Comment (diwako) [inline addons/main/stringtable.xml:1]: Only add new ones or modify existing ones. Deletions are usually not happening unless features are being deleted, in this case for compats, that should not be the case

File: addons/main/initSettings.inc.sqf [hunk 2]
```diff
@@ -74,12 +74,52 @@ private _category = [_header, LLSTRING(subCategoryArmorPlates)];
...
+] call CBA_fnc_addSetting;
+
+[
+    QGVAR(allowPiRReactionsWhilePlatesHoldAI),
+    "CHECKBOX",
+    [LLSTRING(allowPiRReactionsWhilePlatesHoldAI), LLSTRING(allowPiRReactionsWhilePlatesHoldAI_desc)],
+    _category,
+    true,
     true
 ] call CBA_fnc_addSetting;
```
Comment (diwako) [inline addons/main/initSettings.inc.sqf:124]: These are very PIR directed in the general settings. you will most likely need to include them in case PIR is loaded. Do not just remove existing settings. A lot of people are using this without PIR or ACE, that is to be kept in mind

File: addons/main/XEH_preInit.sqf [hunk 3]
```diff
@@ -1,11 +1,18 @@
...
 
-GVAR(aceMedicalLoaded) = isClass(configFile >> "CfgPatches" >> "ace_medical_engine");
-if (isClass(configFile >> "CfgPatches" >> "ace_medical") && {!GVAR(aceMedicalLoaded)}) exitWith {
+private _aceMedicalEngineLoaded = isClass(configFile >> "CfgPatches" >> "ace_medical_engine");
+GVAR(piRLoaded) = isClass(configFile >> "CfgPatches" >> "PiR") || {isClass(configFile >> "CfgPatches" >> "pir_main")};
+GVAR(aceMedicalLoaded) = _aceMedicalEngineLoaded || {GVAR(piRLoaded)};
+
+if (isClass(configFile >> "CfgPatches" >> "ace_medical") && {!_aceMedicalEngineLoaded} && {!GVAR(piRLoaded)}) exitWith {
     INFO("PreInit: Disabled --> old ACE medical loaded");
 };
```
Comment (diwako) [inline addons/main/XEH_preInit.sqf:10]: This is just a hard stop to disallow the old ACE medical from 2018. i doubt you also need to check if pir is not loaded. if someone loads that old medical mod it should just go "no, not doing shit" imo

File: addons/main/XEH_PREP.hpp [hunk 4]
```diff
@@ -32,8 +41,6 @@ if (GVAR(aceMedicalLoaded)) then {
     PREP(commandHeal);
     PREP(disableThirdParty);
     PREP(drawDownedUnitIndicator);
-    PREP(getHitpointArmor);
-    PREP(getItemArmor);
```
Comment (diwako) [inline addons/main/XEH_PREP.hpp:36]: again, careful, these are needed without ACE. there might be a need of a switch between standalone, ace and pir

File: addons/main/XEH_PREP.hpp [hunk 5]
```diff
@@ -4,15 +4,24 @@ PREP(canAddPlate);
...
+PREP(installPiRHandleDamage);
+PREP(installPiRHitPart);
 PREP(moduleHeal);
 PREP(modulePlate);
 PREP(moduleResetMalus);
 PREP(moduleResetMalusGlobal);
+PREP(shouldSuppressPiRReaction);
+PREP(shouldSkipPiRRedirect);
+PREP(piREffectivePlateSettings);
+PREP(pirShansFromHitSelection);
```
Comment (diwako) [inline addons/main/XEH_PREP.hpp:24]: Same with these, only needed if PIR is loaded

File: addons/main/functions/fnc_addPlate.sqf [hunk 6]
```diff
@@ -22,6 +22,10 @@ if (_plates isNotEqualTo []) then {
     _vest setVariable ["ace_movement_vLoad", _vLoad + PLATE_MASS, true];
 };
 _player removeItem QGVAR(plate);
-_vest setVariable [QGVAR(plates), _plates];
+_vest setVariable [QGVAR(plates), _plates, true];
+_player setVariable [QGVAR(plates), _plates, true];
```
Comment (diwako) [inline addons/main/functions/fnc_addPlate.sqf:26]: debug? plates are supposed to be only in the vest

File: addons/main/functions/fnc_addPlate.sqf [hunk 7]
```diff
@@ -22,6 +22,10 @@ if (_plates isNotEqualTo []) then {
...
 [_player] call FUNC(updatePlateUi);
 
+if (_player isEqualTo (call CBA_fnc_currentUnit)) then {
+};
```
Comment (diwako) [inline addons/main/functions/fnc_addPlate.sqf:30]: debug? useless code.

File: addons/main/functions/fnc_addPlateKeyPress.sqf [hunk 8]
```diff
@@ -1,8 +1,12 @@
 #include "script_component.hpp"
 params ["_player"];
 
+GVAR(addingPlate) = true;
```
Comment (diwako) [inline addons/main/functions/fnc_addPlateKeyPress.sqf:4]: that variable is set in the key press handler.- Not sure why you need it here?

**Review by diwako — CHANGES_REQUESTED** [review]

**Comment by diwako** [comment]
on a side note, i might need to update the quality gates for pull requests for the project. It has been a while.

**Comment by x60secAssassin** [comment]
Pivoting to a standalone PiR compat addon (@Armor Plate System PiR Compat) that bundles a plates-only fork + bridge, instead of upstream integration in this PR (PiR has A LOT of conflicting code with APS). Archiving #165 — not abandoning PiR+APS compat, just changing delivery. Thanks for the earlier review/time.

If you are open to it, I will follow up separately about licensing/standalone distribution for the PiR compat pack
